cmd/gateway-main_test.go

```diff
@@ -48,3 +51,34 @@ func TestParseGatewayEndpoint(t *testing.T) {
 		}
 	}
 }
+
+// Test validateGatewayArguments
+func TestValidateGatewayArguments(t *testing.T) {
+	nonLoopBackIPs := localIP4.FuncMatch(func(ip string, matchString string) bool {
+		return !strings.HasPrefix(ip, "127.")
+	}, "")
+	if len(nonLoopBackIPs) == 0 {
+		t.Fatalf("No non-loop back IP address found for this host")
+	}
+	nonLoopBackIP := nonLoopBackIPs.ToSlice()[0]
+
+	testCases := []struct {
+		serverAddr   string
+		endpointAddr string
+		valid        bool
+	}{
+		{":9000", "http://localhost:9001", true},
+		{"123.123.123.123:9000", "http://localhost:9000", false},
+		{":9000", "http://localhost:9000", false},
+		{":9000", nonLoopBackIP + ":9000", false},
```

Comment:
we had a discussion here about this bala with @harshavardhana  , the problem with doing string comparison is the code becomes brittle, i.e if we change the error message string the test case breaks. So I have kept it as "valid bool" for now. But we can have a discussion in one of the meetings so that we can consistently follow one method